File: lenguajesVarios/num.py
import random
import time

# Generar lista de números aleatorios
lista = [random.randint(100, 7000) for _ in range(1000)]
print(lista)
# Medir tiempo de ejecución para ordenamiento (usando el algoritmo de ordenamiento por inserción)
inicio = time.time()
sorted_lista = sorted(lista)
fin = time.time()

tiempo_ejection = (fin - inicio) * 1000  # Convertir a milisegundos
print("Tiempo de ejecución en milisegundos:", tiempo_ejection)

'''
import random
import time

# Función para verificar si una lista está ordenada
def esta_ordenada(lista):
    return all(lista[i] <= lista[i + 1] for i in range(len(lista) - 1))

# Función para realizar Bogosort
def bogosort(lista):
    while not esta_ordenada(lista):
        random.shuffle(lista)
    return lista

# Generar lista de números aleatorios
lista = [random.randint(100, 7000) for _ in range(1000)]

# Medir tiempo de ejecución para Bogosort
inicio = time.time()
lista_ordenada = bogosort(lista)
fin = time.time()

tiempo_ejecucion = (fin - inicio) * 1000  # Convertir a milisegundos
print("Tiempo de ejecución en milisegundos:", tiempo_ejecucion)
'''

File: lenguajesVarios/num.js
// Generar lista de números aleatorios
let lista = Array.from({ length: 1000 }, () => Math.floor(Math.random() * 6901) + 100);

// Medir tiempo de ejecución para ordenamiento (usando el algoritmo de ordenamiento por inserción)
let inicio = performance.now();
lista.sort((a, b) => a - b);
let fin = performance.now();

let tiempo_ejecucion = fin - inicio;
console.log("Tiempo de ejecución en milisegundos:", tiempo_ejecucion);
/*
// Función para verificar si una lista está ordenada
function estaOrdenada(lista) {
    for (let i = 0; i < lista.length - 1; ++i) {
        if (lista[i] > lista[i + 1]) {
            return false;
        }
    }
    return true;
}

// Implementación de Bogosort
function bogosort(lista) {
    while (!estaOrdenada(lista)) {
        lista.sort(() => Math.random() - 0.5);
    }
}

// Generar lista de números aleatorios
let lista = Array.from({ length: 1000 }, () => Math.floor(Math.random() * 6901) + 100);

// Medir tiempo de ejecución para Bogosort
let inicioBogo = performance.now();
bogosort(lista);
let finBogo = performance.now();
let tiempoEjecucionBogo = finBogo - inicioBogo;
console.log("Tiempo de ejecución de Bogosort en milisegundos:", tiempoEjecucionBogo);

// Medir tiempo de ejecución para el segundo algoritmo de ordenamiento (usando el algoritmo de ordenamiento por inserción)
let inicioSegundo = performance.now();
lista.sort((a, b) => a - b);
let finSegundo = performance.now();
let tiempoEjecucionSegundo = finSegundo - inicioSegundo;
console.log("Tiempo de ejecución del segundo algoritmo en milisegundos:", tiempoEjecucionSegundo);

 */